lib/galaxy/datatypes/data.py

```diff
@@ -1116,8 +1120,7 @@ def get_file_peek(file_name, is_multi_byte=False, WIDTH=256, LINE_COUNT=5, skipc
                 line = line[:-1]
                 last_line_break = True
             elif not line_wrap:
-                while True:
-                    i = temp.read(1)
+                for i in file_reader(temp, 1, close=False):
```

Comment:
```suggestion
                for i in file_reader(temp, 1):
```

See proposed changes to `file_reader()` below.